src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php

```diff
@@ -92,6 +92,13 @@ public function load(array $configs, ContainerBuilder $container)
             $dir = dirname($reflection->getFileName()).'/Resources/views';
             if (is_dir($dir)) {
                 $this->addTwigPath($twigFilesystemLoaderDefinition, $dir, $bundle);
+
+                $classMock = new $class();
```

Comment:
`kernel` service is out of scope, so i guess not. Maybe you can kick in at `TwigBundle::createContainerExtension`, without breaking BC.

This is somewhat related to #18563 in terms of improving the ecosystem.